File: app/src/main/java/com/cn29/aac/ui/main/vm/AppArchNavViewModel.java
package com.cn29.aac.ui.main.vm;

import android.content.SharedPreferences;

import androidx.lifecycle.ViewModel;

/**
 * Created by Charles Ng on 13/10/2017.
 */

public class AppArchNavViewModel extends ViewModel {

  SharedPreferences sharedPreferences;

  public AppArchNavViewModel(SharedPreferences sharedPreferences) {
    this.sharedPreferences = sharedPreferences;
  }

  public void logout() {
    sharedPreferences.edit().putBoolean("isLogin", false).apply();
  }
}

File: app/src/main/java/com/cn29/aac/ui/main/vm/AppArchNavViewModel.kt
package com.cn29.aac.ui.main.vm

import android.content.SharedPreferences
import androidx.lifecycle.ViewModel

/**
 * Created by Charles Ng on 13/10/2017.
 */
class AppArchNavViewModel(var sharedPreferences: SharedPreferences) :
        ViewModel() {
    fun logout() {
        sharedPreferences.edit().putBoolean("isLogin", false).apply()
    }

}